### PR TITLE
Apply macro to safe spells (part1)

### PR DESCRIPTION
**Pull request by Hosuke**
cc. @danielpartida for review

### Update!
Please build spells in the proper [subproject](../dbt_subprojects/) directory. For more information, please see the main [readme](../README.md), which also links to a GH discussion with the option to ask questions.

### Contribution type
Please check the type of contribution this pull request is for:

- [ ] New spell(s)
- [x] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

---

### For new spell(s)
If you are building new spell(s), please provide the following information:

- **Spell name(s):** *schema.table_name*
- **Description:** [Detailed description of the new spell(s) and their purpose]
- **Who are the new spell(s) for?** [Internal team or general community]
- **How will the new spell(s) be used downstream?** [Description of downstream usage]
- **Implementation details:** [Information on how the spell(s) are implemented]
- **Test instructions:** [How to test the new spell(s)]
- **Related issue(s):** [Link to related issues, if any]

---

### For adding to existing spell lineage
If you are adding to an existing spell lineage, please provide the following information:

- **Description:** [Description of the changes made]

---

### For bug fixes
If you are fixing a bug, please provide the following information:

- **Description:** [Description of the bug fix]
- **Steps to reproduce:** [How to reproduce the bug]
- **Implementation details:** [Information on how the bug was fixed]
- **Test instructions:** [How to test the fix]
- **Related issue(s):** [Link to related issues, if any]

---

### Additional information
Please provide any additional information that might help us review your pull request:

- [Any additional information]

---

Thank you for your contribution!

### CODE REVIEW

File: dbt_subprojects/hourly_spellbook/models/_project/safe/zksync/safe_zksync_eth_transfers.sql
Comment (Hosuke) [inline dbt_subprojects/hourly_spellbook/models/_project/safe/zksync/safe_zksync_eth_transfers.sql:1]: This spell is `prod_exclude` tagged.

**Comment by jeff-dude** [comment]
thank you for setting the example, @Hosuke 🙏 

fyi @danielpartida we're looking to show an example of how your code can be simplified in spellbook, to make future chain contributions simple. i'll be reviewing and finalizing with commits soon

File: dbt_subprojects/hourly_spellbook/models/_project/safe/optimism/safe_optimism_eth_transfers.sql
Comment (jeff-dude) [inline dbt_subprojects/hourly_spellbook/models/_project/safe/optimism/safe_optimism_eth_transfers.sql:1]: @Hosuke similar to gnosis, did we want to use the macro here and then union the rest? or did it not fit the logic?

File: dbt_subprojects/hourly_spellbook/models/_project/safe/zksync/safe_zksync_eth_transfers.sql
Comment (jeff-dude) [reply dbt_subprojects/hourly_spellbook/models/_project/safe/zksync/safe_zksync_eth_transfers.sql:1]: likely due to duplicates. would need to ensure it's fixed before re-enabling

File: dbt_subprojects/hourly_spellbook/models/_project/safe/optimism/safe_optimism_eth_transfers.sql
Comment (Hosuke) [reply dbt_subprojects/hourly_spellbook/models/_project/safe/optimism/safe_optimism_eth_transfers.sql:1]: The last try did not work well.
I will try again.

Comment (Hosuke) [reply dbt_subprojects/hourly_spellbook/models/_project/safe/optimism/safe_optimism_eth_transfers.sql:1]: Seems working now.

**Comment by jeff-dude** [comment]
@Hosuke can you run some tests to validate data vs. existing? something like:
```
select *
from safe_ethereum.eth_transfers
where block_date between date '2024-08-01' and date '2024-08-18'
except
select *
from <ci_table_ouput>
where block_date between date '2024-08-01' and date '2024-08-18'
```

we should expect to see no results returned, if only thing that changed is macro design. we could do for a few chains to be safe.

**Comment by jeff-dude** [comment]
> @Hosuke can you run some tests to validate data vs. existing? something like:
> 
> ```
> select *
> from safe_ethereum.eth_transfers
> where block_date between date '2024-08-01' and date '2024-08-18'
> except
> select *
> from <ci_table_ouput>
> where block_date between date '2024-08-01' and date '2024-08-18'
> ```
> 
> we should expect to see no results returned, if only thing that changed is macro design. we could do for a few chains to be safe.

here is example i did on another PR which only refactored code:
https://github.com/duneanalytics/spellbook/pull/6343#issuecomment-2293938009

**Comment by Hosuke** [comment]
> > @Hosuke can you run some tests to validate data vs. existing? something like:
> > ```
> > select *
> > from safe_ethereum.eth_transfers
> > where block_date between date '2024-08-01' and date '2024-08-18'
> > except
> > select *
> > from <ci_table_ouput>
> > where block_date between date '2024-08-01' and date '2024-08-18'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > we should expect to see no results returned, if only thing that changed is macro design. we could do for a few chains to be safe.
> 
> here is example i did on another PR which only refactored code: [#6343 (comment)](https://github.com/duneanalytics/spellbook/pull/6343#issuecomment-2293938009)

Sorry seems there are 433 rows of difference:
https://dune.com/queries/4002181

I am checking...

**Comment by Hosuke** [comment]
All matched after refreshed:
https://dune.com/queries/4002181

**Review by jeff-dude — APPROVED** [review]
thank you!

fyi @safeintern 
we are showing an example here of how to leverage macro design to make cross-chain spells easier to build and maintain. this would work for all safe spells 🙏

**Pull request merged**